### PR TITLE
Add tests referenced in project requirements

### PR DESCRIPTION
**Pull request by Chimwemwe-127001**
- Use jest framework for testing
- Create a test for the function that edits the task description.
- Create a test for the function that updates an item's 'completed' status. 
- Test the "Clear all completed" function.
- Mock the HTML elements to test DOM manipulation functions.
- Use the ```describe``` method

> ## In collaboration with @hamid-murambiwa

### CODE REVIEW

File: src/__tests__/todo.test.js [hunk 1]
```diff
@@ -20,36 +20,34 @@ document.body.innerHTML = `
...
+//     },
+//     get length() {
+//       return Object.keys(storage).length;
+//     },
+//     key(i) {
+//       const keys = Object.keys(storage);
+//       return keys[i] || null;
+//     },
+//   };
+// }
```
Comment (EricMbouwe) [inline src/__tests__/todo.test.js:44]: - [ ] Please remove these comments from your code as it makes it look messy.

File: src/__tests__/todo.test.js [hunk 2]
```diff
@@ -20,36 +20,34 @@ document.body.innerHTML = `
...
 
 describe('#addToList is working properly', () => {
   // Arrange
   const todolist = new List();
```
Comment (EricMbouwe) [inline src/__tests__/todo.test.js:48]: - [ ] The tests described in this file are failing. Let's fix that

Screenshot from 2021-11-04 18-59-00

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
### Hi @Chimwemwe-127001,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlight!
- Good PR ✔️
- Good commits ✔️

# Required Changes ♻️ ⛔️

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/__tests__/todo.test.js (hunk 2 above)
Comment (Chimwemwe-127001) [reply src/__tests__/todo.test.js:48]: Hey @EricMbouwe , to fix this issue please run ```npm install .``` or follow instructions in the README file.
And Thanks 😊 for spotting the comments, we will surely get this one fixed ASAP.

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @Chimwemwe-127001 , 

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**